runner/src/server/plugins/builder/pages/summary.js

```diff
@@ -13,66 +14,65 @@ class SummaryViewModel {
 
     const relevantPages = []
     const details = []
-    let endPage = null
 
     let nextPage = model.startPage
-    let repeatCounter = 0
-    let otherRepeatPagesInSection = []
-
-    while (nextPage?.hasFormComponents) {
-      relevantPages.push(nextPage)
-      nextPage = nextPage.getNextPage(state)
-      if(nextPage.repeatField) {
-        const required = reach(state, nextPage.repeatField)
-        if(!otherRepeatPagesInSection.length) {
-          otherRepeatPagesInSection = model.pages.filter(page => page.section === nextPage.section && page.repeatField)
-        }
-        const isEndPage = otherRepeatPagesInSection[otherRepeatPagesInSection.length - 1].path === nextPage.path
-
-        if(isEndPage) {
-          repeatCounter++
-          if(repeatCounter !== required) {
-            relevantPages.push(nextPage)
-            nextPage = relevantPages[relevantPages.length - otherRepeatPagesInSection.length]
-            // repeatCounter = 0
-          } else {
-            repeatCounter = 0
-            otherRepeatPagesInSection = []
-          }
-        }
+    let endPage = null
+
+    while (nextPage != null) {
```

Comment:
can I ask why null and null check

Reply:
null check is because we want to iterate until nextPage isn't null. We can't use `nextPage?.hasFormComponents` because the nextPage won't always have form components, but it's still necessary to continue to walk the whole form.

What do you mean by 'why null'?

Reply:
as opposed to undefined `let endpage`. Just curious